cloud_blobstore/s3.py

```diff
@@ -277,3 +278,30 @@ def find_next_missing_parts(
                 if parts_resp['IsTruncated'] and search_start == parts_resp['NextPartNumberMarker']:
                     # finished examining the results of this batch, move onto the next one
                     break
+
+    def check_bucket_exists(self, bucket: str) -> bool:
```

Comment:
I think this is a bit of a misnomer.  This tests if the bucket is accessible, not if it exists.

Reply:
if bucket is not accessible I'm also checking the error code.
If it was a 404 error, then the bucket does not exist.

Reply:
Ah you are right.  I misread the documentation!

Thanks!